### ISSUE TITLE
Sunsetting support for data-vocabulary in April 2020

### ISSUE THREAD

**Issue by GATAKAWAKACHICO** [kzkn/gretel#16]
Hi, thank you very much for developing such a cool gem.

I have used breadcrumb's semantic option. I received a warning that data-vocabulary will be deprecated from Google Search Console yesterday.

As I mentioned title, data-vocabulary are sunsetting.
https://webmasters.googleblog.com/2020/01/data-vocabulary.html

I think this gem requires to be fixed, so I researched which codes need to be modified:

At render_semantic_fragment function, 

https://github.com/WilHall/gretel/blob/260a4c1e02ba314447f382601c9e49943734c61b/lib/gretel/renderer.rb#L211

- Add itemtype "https://schema.org/BreadcrumbList" to container_tag
- Change itemprop of li tag to itemListElement, itemtype to "https://schema.org/ListItem"
- Chnage itemprop of a tag from url to item
- Change itemprop of span tag from title to name

At render function, 

https://github.com/WilHall/gretel/blob/260a4c1e02ba314447f382601c9e49943734c61b/lib/gretel/renderer.rb#L174

- Add position of ListItem using map.with_index

Could anyone please give me a feedback?

**Comment by StarWar** [comment]
I've forked the parent repository from lassebunk/gretel and edited the renderer.rb... for simple and basis implementation it seems to work

 @https://github.com/StarWar/gretel/blob/master/lib/gretel/renderer.rb

**Comment by GATAKAWAKACHICO** [comment]
@StarWar I am very impressed with your work. 

Have you tried the test as below? First, creating branch will be needed. If it will pass, other people can use your feature!
https://github.com/WilHall/gretel#contributing

**Comment by StarWar** [comment]
@GATAKAWAKACHICO I've made a slight change to the renderer (forgot to check for the semantic option for wrapping div) and some relevant changes to tests (replacing data-vocab with schema.org and relevant changes to existing tests) and now tests are passing. I haven't added more tests to it.

**Comment by morgant** [comment]
@StarWar Please do create a feature branch, I'd like to test this too.

**Comment by StarWar** [comment]
@morgant just created a branch feature/schema-org-breadcrumb

**Comment by GATAKAWAKACHICO** [comment]
@StarWar 

> I've made a slight change to the renderer (forgot to check for the semantic option for wrapping div) and some relevant changes to tests (replacing data-vocab with schema.org and relevant changes to existing tests) and now all existing tests are passing. I haven't added more tests to it though.

Cool! Could you please open the pull request of those changes(feature/schema-org-breadcrumb)? Reviewing will be required and after that it can be merged.

**Comment by StarWar** [comment]
@GATAKAWAKACHICO I've created a pull-request for the main branch @lassebunk/gretel This @WilHall/gretel has some conflicts though. Should I still create a pull request here as well?

**Comment by GATAKAWAKACHICO** [comment]
@StarWar I think it's better that you also create a pull request on @WilHall/gretel, because this repository is maintained from @lassebunk/gretel, which is no longer maitained according to this [link](https://github.com/lassebunk/gretel#this-repository-is-no-longer-maintained) . 

However, [the latest gem](https://rubygems.org/gems/gretel/versions/3.0.9) still refers @lassebunk/gretel. 

I am hoping @WilHall, @MrShemek who are the owner of this repo give us some advice. @WilHall, @MrShemek How do you think about merging this new feature?

**Comment by morgant** [comment]
@StarWar Thanks again for implementing your changes and putting them in a feature branch. I haven't run tests yet, but I did a quick code review of https://github.com/lassebunk/gretel/compare/master...StarWar:feature/schema-org-breadcrumb and the changes in `lib/gretel/renderer.rb` look good.

I second @GATAKAWAKACHICO's request to create a PR for this repo as it _appears_ to be the one that is maintained. My guess is that @WilHall & @MrShemek will prefer the commit to bump the version number to be excluded as they'll do that when ready for a release. I'm not sure if the changes to the schema should be moved into a separate PR or not.

**Comment by WilHall** [comment]
@GATAKAWAKACHICO @morgant 

Unfortunately @lassebunk passed away shortly after I had talked with him about helping to maintain the gem. When @MrShemek and I created the fork we were hoping to be able to keep this gem going, at least for maintenance.

I haven't heard from @MrShemek in awhile now, and unfortunately I have not been doing much ruby work as of late, and so I haven't had the time to put in the maintenance that is required.

If anyone would like to take over responsibility for getting a new release out with this fix and with others, that would be appreciated. I'm happy to help insofar as transferring ownership and facilitating a transition, but I don't have much time to dedicate to maintenance right now.

**Comment by GATAKAWAKACHICO** [comment]
I appreciate @StarWar's really quick contribution again, and @WilHall's cooperation so far. In addition, I also understand his current situation. I think that it is up to the individual to contribute to OSS activities. I just suppose this repo's travis-ci environment is unfortunately very old, so it's better to be fixed that everyone freely creates PRs, but who are able to maintain...?

Anyway I confirmed it works without errors on my existing Docker rails source by using the branch.

In Gemfile, 

```ruby:Gemfile.rb
gem 'gretel', github:'StarWar/gretel', branch: 'feature/schema-org-breadcrumb'
```

You may need to type a command as below.

```bash:bundle.sh
bundle install --path vendor/bundle
```

And please be careful that [Rich results tool](https://search.google.com/test/rich-results) doesn't recognize relative urls; but it doesn't have a negative effect if you use them in production environment.
https://webuniverse.io/relative-urls-in-structured-data/

**Comment by MrShemek** [comment]
I volunteered to help @WilHall with maintaining this gem as a part of my university project. Shortly after that, I decided to drop studies and to be honest, I am no longer using Gretel (different job, different projects).

Unfortunately, I will not have time to maintain this gem. Good luck guys!

**Comment by morgant** [comment]
@WilHall & @MrShemek Many thanks for taking the time to respond and I too completely understand your situation and! I'm not sure I have the availability to take over maintenance of this gem either as I'm only using it for one client, but will do what I can to help with this issue. Hopefully we can find someone who's able to take over.

I've identified several similar PRs to @lassebunk/gretel (lassebunk#86, lassebunk#83, and lassebunk#59) which should be compared to @StarWar's (lassebunk#105) and closed once we have a final implementation.

**Comment by morgant** [comment]
I have cherry-picked @StarWar's commits from lassebunk#105 into this repo's `develop` branch, less the version bump and some other formatting changes (to keep it as straightforward a merge as possible). See PR #17. Hopefully this'll make it easier for us to get this merged and move forward.

Thanks again, @StarWar!

**Comment by GATAKAWAKACHICO** [comment]
@morgant Thank you for cherry-picking.
On Travis CI, sqlite adapter's problem occurred.

https://travis-ci.org/WilHall/gretel/jobs/657980931#L539
https://travis-ci.org/WilHall/gretel/jobs/657980932#L442
https://travis-ci.org/WilHall/gretel/jobs/657980936#L435

I created same environment as the CI on Docker. By specifying sqlite version of Gemfile-rails.x.x.x, it works.
`gem "sqlite3", "~> 1.3.6"`

Finally, #18 have passed all checks. This branch has merged #17's changes.

**Comment by morgant** [comment]
@GATAKAWAKACHICO Many thanks!

**Comment by takakii** [comment]
@GATAKAWAKACHICO 
What is the current status of this issue?
Will it be merged?

**Comment by GATAKAWAKACHICO** [comment]
@takakii Still ongoing.

@morgant Could you please review #18 as long as you're not busy? Review required to merge it, and I think you are the right person to do it because you have seen this issue before.

**Comment by GATAKAWAKACHICO** [comment]
Hi, there!
Sunsetting of data-vocabulary postponed because of COVID-19, but they says re-evaluate this matter in June 2020. 
https://webmasters.googleblog.com/2020/01/data-vocabulary.html
As a thank you to everyone involved, I think it would be better to merge #18 into this gem.
Someone please help!!

**Comment by kzkn** [comment]
I'm following this issue. I have commented #17 .

I'm planning to release 4.0.0. This release basically adds no new features (except for those already committed to the develop branch) and is intended to drop support for older Ruby and Rails and remove deprecated code.

Resolution of this issue will likely come after that release, probably in 4.1.0 or later.

**Comment by kzkn** [comment]
I have shipped this fix in 4.0.1. Give it a try!
